### PR TITLE
feat: Scan-Add a book

### PR DESCRIPTION
**Pull request by shadowciaw**
#### Summary of changes/features added:
- added scanning book feature to add a book

#### Screenshots of UI changes (update with each new commit): 
image

## Pull Request Checklist
- [ ] Unit test written?
- [ ] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

**Comment by shadowciaw** [comment]
will need to convert button to app's button later

**Review by rwmilford — COMMENTED** [review]
Looks good, but to be fair this isn't something I know a great deal about.  From what I do understand of it, it makes sense.

File: client/app/src/main/res/layout/activity_scan.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <Button
+        android:id="@+id/scan_confirm_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="16dp"
+        android:text="@string/scan_confirm"
```
Comment (AdityaHarvi) [inline client/app/src/main/res/layout/activity_scan.xml:27]: Buttons aren't formatted properly 😛 
Should be using the background and non-cap text

**Review by AdityaHarvi — CHANGES_REQUESTED** [review]

File: client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java [hunk 2]
```diff
@@ -115,4 +138,34 @@ public void onBackPressed() {
         startActivity(intent);
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java:142]: java docs

File: client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java [hunk 3]
```diff
@@ -41,15 +53,17 @@ protected void onCreate(Bundle savedInstanceState) {
         final Button saveButton = findViewById(R.id.newBookSaveButton);
         final Button scanButton = findViewById(R.id.newBookScanButton);
 
+
```
Comment (rjmah1) [inline client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java:56]: what's with these double spaces?

File: client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java [hunk 4]
```diff
@@ -115,4 +135,39 @@ public void onBackPressed() {
...
+                Log.d("testing", "barcode data=" + barcodeData);
+
+                try {
+                    Book book = new GoogleBooksWrapper().getBook(barcodeData);
+                    Log.d("testing book", "title");
+                    bookTitle.setText(book.getTitle());
+                    bookAuthor.setText(book.getAuthor());
+                    bookDesc.setText(book.getDescription());
+                    bookISBN.setText(book.getIsbn());
+                } catch (IOException e) {
```
Comment (rjmah1) [inline client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java:162]:
```suggestion
                } catch (Exception e) {
```

File: client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java (hunk 3 above)
Comment (shadowciaw) [reply client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java:56]: idk im just crazy like that (fixed)

File: client/app/src/main/java/com/example/criengine/Activities/ScanActivity.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        // Camera permission is required to start the scan
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA}, REQUEST_PERMISSION_CAMERA);
+            return;
+        }
+        initialize();
+    }
+
+    /**
+     *  Once the user clicks yes or no on granting the app camera access, process the result here.
```
Comment (rjmah1) [inline client/app/src/main/java/com/example/criengine/Activities/ScanActivity.java:66]: what's that extra space doin in the javadoc?

**Review by AdityaHarvi — APPROVED** [review]

File: client/app/src/main/java/com/example/criengine/Activities/ScanActivity.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                                barcodeData = barcodes.valueAt(0).email.address;
+                            } else {
+                                barcodeData = barcodes.valueAt(0).displayValue;
+                            }
+                            // once barcode data obtained, enable the confirmation button to return to previous activity with barcode data.
+                            barcodeText.setText(R.string.scan_complete);
+                            confirmScanButton.setEnabled(true);
+                        }
+                    });
+
```
Comment (rjmah1) [inline client/app/src/main/java/com/example/criengine/Activities/ScanActivity.java:174]: this empty space

**Review by rjmah1 — CHANGES_REQUESTED** [review]
code quality matters

**Review by rjmah1 — APPROVED** [review]
lgtm

**Pull request merged**